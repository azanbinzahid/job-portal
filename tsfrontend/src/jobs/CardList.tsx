import React, { FC } from "react";
import styled from "styled-components";
import { Card, ListGroup } from "react-bootstrap";

const Title = styled.div`
  color: white;
`;

type Props = {
  title: string;
  list: string[];
};

const CardList: FC<Props> = ({ title, list }) => {
  console.log(list);
  return (
    <Card bg="primary">
      <Card.Header>
        <Title>{title}(s) </Title>
      </Card.Header>
      <ListGroup variant="flush">
        {list
          ? list.map((ele, index) => (
              <ListGroup.Item key={index}>{ele}</ListGroup.Item>
            ))
          : null}
      </ListGroup>
    </Card>
  );
};

export default CardList;
