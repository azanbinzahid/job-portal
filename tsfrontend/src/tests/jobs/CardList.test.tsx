import React from "react";
import { shallow } from "enzyme";
import CardList from "jobs/CardList";
test("renders the component", () => {
  const component = shallow(<CardList title="" list={[]} />);
  expect(component).toMatchSnapshot();
});
