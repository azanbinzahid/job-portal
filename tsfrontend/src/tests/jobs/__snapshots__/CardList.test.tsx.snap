// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Card
  bg="primary"
  body={false}
>
  <CardHeader>
    <styled.div>
      (s) 
    </styled.div>
  </CardHeader>
  <ListGroup
    variant="flush"
  />
</Card>
`;
